gnome-initial-setup/pages/account/um-photo-dialog.c

```diff
@@ -169,13 +169,40 @@ menu_item_for_filename (UmPhotoDialog *um,
         return menuitem;
 }
 
+static GSList *
+get_facesdirs (void)
+{
+        GSList *facesdirs = NULL;
+        const gchar * const * data_dirs;
+        int i;
+        const char *facesdir_env;
+
+        facesdir_env = g_getenv ("GIS_ACCOUNT_PAGE_FACESDIR");
+        if (facesdir_env != NULL && g_strcmp0 (facesdir_env, "") != 0) {
+                facesdirs = g_slist_prepend (facesdirs,
+                                             g_strdup (facesdir_env));
+        }
+
+        if (g_strcmp0 (ACCOUNT_PAGE_FACESDIR, "") != 0) {
+                facesdirs = g_slist_prepend (facesdirs,
+                                             g_strdup (ACCOUNT_PAGE_FACESDIR));
+        }
+
+        data_dirs = g_get_system_data_dirs ();
+        for (i = 0; data_dirs[i] != NULL; i++) {
+                char *path = g_build_filename (data_dirs[i], "pixmaps", "faces", NULL);
+                facesdirs = g_slist_prepend (facesdirs, path);
+        }
+
+        return g_slist_reverse (facesdirs);
```

Comment:
If you build the list with `GQueue` (which maintains a tail pointer) you don't need to do the "prepend then reverse" dance.

Reply:
Unless you feel strongly about this, I will leave it with a list for now since it's just a few items.

Reply:
No, I don't feel strongly about it. (I have felt weakly but persistently about it since 2011, when I wrote a blog post about it that was lost in a database crash, but that didn't stop “Bob Bobson” [critiquing it](https://blogs.gnome.org/wjjt/2011/03/13/emusic-in-banshee/#comment-14).)